### PR TITLE
Added support for request body inflation and added relevant tests

### PR DESCRIPTION
**Pull request by bminer**
Added support for request body inflation and added relevant tests
Release 4.1.0
This closes issue #38.

**Please do not merge this PR** until the following inflation pull request is merged and published to NPM.  1 test will fail until this PR is merged.
https://github.com/stream-utils/inflation/pull/14

### CODE REVIEW

**Comment by bminer** [comment]
The Travis CI builds will fail because this PR references inflation@1.1 in the package.json file, which has not yet been published.  Please wait for https://github.com/stream-utils/inflation/pull/14 to be merged and published to NPM before merging this PR.  Thanks and sorry for the confusion.

**Comment by bminer** [comment]
One bug that I noticed is that the `Content-Length` header is passed to raw-body, which causes a body length mismatch whenever a `Content-Encoding` header is provided.  This needs to be fixed.

**Comment by bminer** [comment]
This can now be safely merged.  All tests passing.

File: History.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+4.1.0 / 2016-03-21
+==================
+
+  * Added support for request body inflation
```
Comment (fengmk2) [inline History.md]: No need to change history, this will be change before npm publish.

File: test/any.js [hunk 2]
```diff
@@ -147,4 +148,57 @@ describe('parse(req, opts)', function(){
...
+
+  describe('with content-encoding', function(){
+    it('should inflate gzip', function(done){
+      var app = koa();
+      var json = JSON.stringify({ foo: 'bar' });
+
+      app.use(function *(){
+        var body = yield parse(this);
+        body.should.eql({ foo: 'bar' });
+        done();
```
Comment (fengmk2) [inline test/any.js]: set parse body to `this.body` and assert body on `request.expect(200, { foo: 'bar' })`

**Comment by fengmk2** [comment]
@bminer wait for you to fix these, and then I will merge and release a new minor version.

File: test/any.js (hunk 2 above)
Comment (bminer) [inline test/any.js]: @fengmk2 - I don't understand.  The server should be verifying the client's request body.  Why should this be changed?

**Comment by fengmk2** [comment]
Landed b00173441920ee65b1691cd8fd18d51a9d043107

**Comment by fengmk2** [comment]
4.2.0

**Comment by bminer** [comment]
Thank you!

**Pull request closed without merging**